noticeboardapp/views.py

```diff
@@ -0,0 +1,158 @@
+from django.shortcuts import render, redirect
+from datetime import date, datetime
+from noticeboardapp.models import Notice
+from django.contrib.auth.decorators import login_required
+
+
+
+# Create your views here.
+
+
+@login_required
+def add_notice_view(request):
+
+
+    if request.method == "GET":
+
+        today = date.today()
+        current_day = today.strftime("%A")
+
+        return render(
+            request,
+            'noticeboard/add_notice.html',
+            context={
+                'today': today,
+                'current_day': current_day,
+            }
+        )
+
+    if request.method == "POST":
+
+        title = request.POST.get('title')
+        notice = request.POST.get('notice')
+        current_user = request.user
+        timestamp = datetime.now()
+
+        if notice:
+            Notice.objects.create(content=notice, timestamp=timestamp, title=title, user=current_user)
+
+            notices = Notice.objects.all().order_by('-timestamp')
+            today = date.today()
+            current_day = today.strftime("%A")
+
+            return render(
+                request,
+                'noticeboard/board.html',
+                context={
+                    'notices': notices,
+                    'today': today,
+                    'current_day': current_day,
+                    'current_user': current_user,
+                }
+            )
+
+        return redirect('noticeboardapp:noticeboard-view')
+
+
+@login_required
+def noticeboard_view(request):
+
+    notices = Notice.objects.all().order_by('-timestamp')
+    current_user = request.user
+    today = date.today()
+    current_day = today.strftime("%A")
+
+    if request.method == "GET":
```

Comment:
this condition and else is not necessary. 
@login_required decorator checks permission.